From 3830ffa1843b8f1819f88971be35308d1ff165be Mon Sep 17 00:00:00 2001
From: skongkonga <3139771198@qq.com>
Date: Fri, 18 Oct 2024 12:06:22 +0800
Subject: [PATCH] feat: add PlayerTitleEvent


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index ab5d28f74..89b4b8fd9 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -66,6 +66,7 @@ import org.bukkit.event.player.PlayerGameModeChangeEvent;
 import org.bukkit.event.player.PlayerReceiveMessageEvent;
 import org.bukkit.event.player.PlayerRegisterChannelEvent;
 import org.bukkit.event.player.PlayerTeleportEvent;
+import org.bukkit.event.player.PlayerTitleEvent;
 import org.bukkit.event.player.PlayerUnregisterChannelEvent;
 import org.bukkit.inventory.InventoryView.Property;
 import org.bukkit.map.MapCursor;
@@ -1669,16 +1670,21 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     @Override
     public void sendTitle(String title, String subtitle, int fadeIn, int stay, int fadeOut) {
-        PacketPlayOutTitle times = new PacketPlayOutTitle(fadeIn, stay, fadeOut);
+        PlayerTitleEvent playerTitleEvent = new PlayerTitleEvent(this, title, subtitle, fadeIn, stay, fadeOut);
+        server.getPluginManager().callEvent(playerTitleEvent);
+        if (playerTitleEvent.isCancelled())
+            return;
+
+        PacketPlayOutTitle times = new PacketPlayOutTitle(playerTitleEvent.getFadeIn(), playerTitleEvent.getStay(), playerTitleEvent.getFadeOut());
         getHandle().playerConnection.sendPacket(times);
 
         if (title != null) {
-            PacketPlayOutTitle packetTitle = new PacketPlayOutTitle(EnumTitleAction.TITLE, CraftChatMessage.fromString(title)[0]);
+            PacketPlayOutTitle packetTitle = new PacketPlayOutTitle(EnumTitleAction.TITLE, CraftChatMessage.fromString(playerTitleEvent.getTitle())[0]);
             getHandle().playerConnection.sendPacket(packetTitle);
         }
 
         if (subtitle != null) {
-            PacketPlayOutTitle packetSubtitle = new PacketPlayOutTitle(EnumTitleAction.SUBTITLE, CraftChatMessage.fromString(subtitle)[0]);
+            PacketPlayOutTitle packetSubtitle = new PacketPlayOutTitle(EnumTitleAction.SUBTITLE, CraftChatMessage.fromString(playerTitleEvent.getSubtitle())[0]);
             getHandle().playerConnection.sendPacket(packetSubtitle);
         }
     }
-- 
2.42.0.windows.2

